### ISSUE TITLE
[FEATURE] Disable "Automate" button when test automation is at 100%

### ISSUE THREAD

**Issue by ChowSingYeung** [COMP3122-2324-Project/Simpsyber#31]
In the "Test Automation" menu, we can "Automate" or "Fix" the tests, it is expected that when there is no error, we cannot press the "Fix" button, but when testcases are 100% automated, we can still press the "Automate" button, assign a staff and press "OK", and the task will finish immediately even if the game is pasued. This is a meaningless action and should be eliminated as it may create confusion to the player.

To fix this, the "Automate" button should be disabled just like the "Fix" Button when test automation is at 100%.

https://github.com/COMP3122-2324-Project/Simpsyber/assets/115892724/0fbd8b9c-fff5-4587-80ac-2e48eb99206f

**Comment by TANG-Yu-Hin-23041117d** [comment]
Yeah, I agree with you. The button should be disallowed when the completion is 100%. Otherwise, users are easy to get confused. Besides that, I suggest that each time the upgrade is complete, the person assigned to the update should be released automatically rather than the user removing them manually.

**Comment by Kiawaik** [comment]
I agree the suggestion. The Automate button shouldn't be clickable when it is 100%. The Automate button should be disabled and turned to grey colour to alert the player that it is not available since it reach the max 100%. 
Otherwise, it is strange and meaningless because the player can click it, but the player's action can not be executed inside since it already gets 100%.

**Comment by ChowSingYeung** [comment]
> Yeah, I agree with you. The button should be disallowed when the completion is 100%. Otherwise, users are easy to get confused. Besides that, I suggest that each time the upgrade is complete, the person assigned to the update should be released automatically rather than the user removing them manually.

Yeah but would you mind elaborate more on the automatical remove part? I thought the system removes the assigned staff automatically already when a task is finished.

**Comment by TANG-Yu-Hin-23041117d** [comment]
> > Yeah, I agree with you. The button should be disallowed when the completion is 100%. Otherwise, users are easy to get confused. Besides that, I suggest that each time the upgrade is complete, the person assigned to the update should be released automatically rather than the user removing them manually.
> 
> Yeah but would you mind elaborate more on the automatical remove part? I thought the system removes the assigned staff automatically already when a task is finished.

I am so sorry that I made the mistake that it was released automatically. I found another enhancement. The percentage of showing automatic should be accurate. The video shows the automatic rate is 99% in detail, but it shows 100% in the panel. Users may not easily notice it, and may affect their final result.


https://github.com/COMP3122-2324-Project/Simpsyber/assets/157493714/6c2279c6-92da-42d8-995f-fba706f1a35e

**Comment by ChowSingYeung** [comment]
> > > Yeah, I agree with you. The button should be disallowed when the completion is 100%. Otherwise, users are easy to get confused. Besides that, I suggest that each time the upgrade is complete, the person assigned to the update should be released automatically rather than the user removing them manually.
> > 
> > 
> > Yeah but would you mind elaborate more on the automatical remove part? I thought the system removes the assigned staff automatically already when a task is finished.
> 
> I am so sorry that I made the mistake that it was released automatically. I found another enhancement. The percentage of showing automatic should be accurate. The video shows the automatic rate is 99% in detail, but it shows 100% in the panel. Users may not easily notice it, and may affect their final result.
> 
>  Unity.WebGL.Player._.SimPsyber.and.4.more.pages.-.Personal.-.Microsoft.Edge.2024-03-04.00-46-19.mp4

I actually found this issue as well and created a post for this. What is more confusing is that we cannot know which data is accurate as in my case one 99% test can be automated but the other one cannot. See the video in the post.
https://github.com/COMP3122-2324-Project/Simpsyber/issues/30
